apps/verticalhome/js/pinned_apps_navigation.js

```diff
@@ -13,274 +13,375 @@
+      ){
+        scrollElemNum = scrollElemNum % ElsLength;
       }
 
-      var topStyle = null;
+      var step = parseInt(window.getComputedStyle(this.elements[0])
+        .getPropertyValue('height'));
 
-      switch(e.key) {
-        case 'ArrowUp':
+      var top_pos = window.getComputedStyle(this.elemList,null)
+        .getPropertyValue('top');
 
-          preVerticalNavigation();
-          document.getElementById('clock').classList.add('not-visible');
+      var fix = 110;
 
-          if (this.elemList.dataset.scrollup) {
-            this.elemList.style.top = this.elemList.dataset.scrollup;
-          } else {
-            this.elemList.style.top = '-6rem';
-            this.elemList.dataset.scrollup = this.elemList.style.top;
-          }
+      if(scrollElemNum<0){
+        fix = 0;
+      }
 
-          var deltaElemsToStartScroll = 3;
+      this.elemList.classList.add('animation');
+      this.elemList.style.top =
+        parseInt(top_pos)-((step+this.stepKoef)*scrollElemNum)+fix+'px';
 
-          this.elements[this.selectedElemIndex].classList.remove('selected');
-          if (this.selectedElemIndex >= 0 && this.selectedElemIndex < 4) {
+      return true;
+    },
 
-            var cycledElem = deltaElemsToStartScroll - this.selectedElemIndex;
-            cycleElements(cycledElem);
-            this.selectedElemIndex += cycledElem;
+    afterAnimation: function afterAnimation(){
+      var pinnedAppsNavElList = app.pinnedAppsNavigation.elemList;
+      pinnedAppsNavElList.classList.remove('animation');
+      pinnedAppsNavElList.style.top = app.pinnedAppsNavigation.defTopval;
 
-            this.selectedElemIndex--;
-            this.elements[this.selectedElemIndex].classList.add('selected');
+       return true;
+    },
+    /**
+     * Handle event keydown
+     * @param  {[Object]} e [default object]
+     * @return {[nothing]}   [nothing]
+     */
+    handleEvent: function(e) {
+      var mainscreen = document.getElementById('main-screen');
+      this.personalise_mode = mainscreen.classList.contains('personalize_mode');
+      this.rearrange_mode = mainscreen.classList.contains('rearrange_mode');
 
-          } else {
+      if (this.preventDef) {
+        e.preventDefault();
+      }
 
-            this.selectedElemIndex--;
-            this.elements[this.selectedElemIndex].classList.add('selected');
+      var currSelected =  null;
+      var nextSelected = null;
+      var prevSelected = null;
+      var needClone = true;
+      var itm = null;
+      var cln = null;
 
-            if (this.selectedElemIndex > 3) {
-              topStyle = parseFloat(this.elemList.style.top);
-              this.elemList.style.top =  topStyle + 6 + 'rem';
-              this.elemList.dataset.scrollup = this.elemList.style.top;
-            }
+      switch(e.key) {
+        case 'ArrowUp':
+          e.preventDefault();
+          document.getElementById('clock').classList.add('not-visible');
```

Comment:
Implement Clock.show()/hide() methods. I think you should not directly change this element.